contracts/test-integration/BlsProvider.test.ts

```diff
@@ -411,6 +411,183 @@ describe("BlsProvider", () => {
     expect(chainId).to.equal(expectedChainId);
     expect(accounts).to.deep.equal(expectedAccounts);
   });
+
+  it("should return the number of transactions an address has ever sent", async function () {
+    // Arrange
+    const address = await blsSigner.getAddress();
+    const expectedTransactionCount = await regularProvider.getTransactionCount(
+      address,
+    );
+    const expectedTransactionCountAtEaliestBlockTag =
+      await blsProvider.getTransactionCount(address, "earliest");
+
+    // Act
+    const transactionCount = await blsProvider.getTransactionCount(address);
+    const transactionCountAtEaliestBlockTag =
+      await blsProvider.getTransactionCount(address, "earliest");
+
+    // Assert
+    expect(transactionCount).to.equal(expectedTransactionCount);
```

Comment:
Why is the `blsProvider` expected to give the same transaction count as `regularProvider`?

Shouldn't `blsProvider` be calling `nonce()` on the contract? And `regularProvider` would just return zero I suppose? Or error?

Reply:
Does `blsSigner.getAddress()` return an EOA account somehow? I'd expect a `BLSWallet`.

Reply:
Ah yeah missed this completely. Because blsProvider was just calling the ethers version of getTransactionCount, both calls were returning 0.

> Does blsSigner.getAddress() return an EOA account somehow? I'd expect a BLSWallet.
blsSigner.getAddress() does [return the bls wallet address](https://github.com/web3well/bls-wallet/blob/db968c34fd7ce30e01e0a3048aa9075370d367a0/contracts/clients/src/BlsSigner.ts#L98-L106). 

Added a new `getTransactionCount` method onto the blsProvider itself that follows similar logic to BlsWalletWrapper, with the addition of being able to query by block tag.